### DISCUSSION TITLE
Collections Schema: Possibility to add more vocabulary to JSON schema

### DISCUSSION THREAD

**Discussion by hhmric** [ldproxy/ldproxy#731] [Configuration]
Hi, 
I have a question regarding the Collections Schema: Is there any possibility or future plan to enable the usage of more vocabulary than the one provided in JSON Schema 2019-09? It would be interesting to add other existing vocabularies or own keywords in order to include more metadata. Thank you very much for your help!

**Comment by cportele** [comment]
Currently there are no options for this. This building block is currently waiting for progress in the [standardization in OGC](https://github.com/opengeospatial/ogcapi-features/projects/11). 

Could you perhaps describe in more detail or provide an example what you have in mind? Thanks.

**Reply by hhmric** [reply]
Thanks for your answer! 
I was wondering whether it would be possible to add more information on the attributes to the JSON, in particular the unit of an attribute. In this way, all relevant information on the attribute (name, description, type, unit) would be available in the JSON and can be used easily in further applications.

**Reply by hhmric** [reply]
Or would it be possible to use the unit value that can be configured in the feature provider? (https://docs.ldproxy.net/providers/#types)

**Reply by cportele** [reply]
Yes, we recently added the "unit" attribute to the feature provider schema. It will be included in the next ldproxy release 3.3.0 (planned for December).

In general, it would be simple to also write the unit as a extension keyword in the schema. However, extending the JSON Schema vocabulary will only be effective, if client tooling supports the extensions. 

A simpler approach that we typically use and that requires no extensions is to include the unit in the label and/or description. For example, use "duration [s]" as the attribute label instead of "duration".

Of course, one can also do both (include the unit in the label and extend the vocabulary).

**Reply by azahnen** [reply]
FYI, propagation of provider units to JSON schema was merged into master, see #781.
